src/app/shared/services/base.service.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+
+import { environment } from '../../../environments/environment';
+import { BaseService } from './base.service';
+import { HttpErrorResponse } from '@angular/common/http';
+
+describe('BaseService', () => {
+  let service: BaseService;
+  let httpMock: HttpTestingController;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule
+      ]
+    });
+    service = TestBed.inject(BaseService);
+    httpMock = TestBed.inject(HttpTestingController);
+  });
+
+  it('creates the service', () => {
+    expect(service).toBeTruthy();
+  });
+
+  describe('#post', () => {
+    describe('Given a valid data', () => {
+      it('returns an Observable<any>', () => {
+        const mockResponse = {
+          data: {
+            id: 10,
+            type: 'token',
+            attributes: {
+              access_token: 'lbxD2K2BjbYtNzz8xjvh2FvSKx838KBCf79q773kq2c',
+              token_type: 'Bearer',
+              expires_in: 7200,
+              refresh_token: '3zJz2oW0njxlj_I3ghyUBF7ZfdQKYXd2n0ODlMkAjHc',
+              created_at: 1597169495
+            }
+          }
+        };
+
+        const deserializedResponse = {
+          id: 10,
+          accessToken: 'lbxD2K2BjbYtNzz8xjvh2FvSKx838KBCf79q773kq2c',
+          tokenType: 'Bearer',
+          expiresIn: 7200,
+          refreshToken: '3zJz2oW0njxlj_I3ghyUBF7ZfdQKYXd2n0ODlMkAjHc',
+          createdAt: 1597169495
+        };
+
+        const data = {
```

Comment:
I was a bit confused by the naming `data` as I thought initially it referred to the data attributethe JSON response. So I would recommend to make this variable name more specific e.g. `postRequestPayload`.